### PR TITLE
Add Rust publish workflow

### LINKED ISSUE #869: [Task] Release to crates.io
## Description
In order to release to crates.io we need rename our packages and setup the automatic release process.

## To-do list

- [x] Rename crates - #895
- [x] Exclude `examples` (see https://github.com/iotaledger/identity.rs/runs/6837111260?check_suite_focus=true where examples would be published) - #895
- [x] Publish crates.
- [ ] setup CI to push automatically

### PR DESCRIPTION
**Pull request by eike-hass**
# Description of change
Add manual Rust release job for publishing to crates.io.

## Links to any relevant issues
Closes #869.

## Type of change
Add an `x` to the boxes that are relevant to your changes.

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [X] Enhancement (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Fix

## How the change has been tested
Tested the command locally, job can only be tested on github after we merge it to `dev`.

## Change checklist
Add an `x` to the boxes that are relevant to your changes.

- [X] I have followed the contribution guidelines for this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/rust-release-to-crates.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: Rust release to crates.io
```
Comment (cycraig) [inline .github/workflows/rust-release-to-crates.yml:1]: Rename this to `rust-publish`. We use the "release" terminology separately.

```suggestion
name: Rust public to crates.io
```

**Review by cycraig — COMMENTED** [review]
Minor changes.

File: .github/workflows/rust-release-to-crates.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.inputs.branch }}
+      - name: Release to npm
+        uses: './.github/actions/publish/publish-rust'
+        with:
+          version: ${{ github.event.inputs.version }}
+          crates-token: ${{ secrets.CRATES_IO_TOKEN }}
+          dry-run: ${{ github.event.inputs.version && 'true' }}
```
[the remark this answers is not in the document]
Comment (eike-hass) [reply .github/workflows/rust-release-to-crates.yml:31]: I mean, what is the worst that could happen? Oh right... :D

**Review by cycraig — APPROVED** [review]
Looks good to me. Do we want to add this to the `rust-automatic-release-and-publish` workflow yet or wait for manual confirmation first? We'll need the manual action either way for dry-runs.

**Comment by eike-hass** [comment]
> Looks good to me. Do we want to add this to the `rust-automatic-release-and-publish` workflow yet or wait for manual confirmation first? We'll need the manual action either way for dry-runs.

I would suggest we wait till after the first successful release and then decide if we want to fully automate.

**Pull request merged**